packages/block-editor/src/components/block-toolbar/use-has-block-toolbar.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { getBlockType, hasBlockSupport } from '@wordpress/blocks';
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { useHasAnyBlockControls } from '../block-controls/use-has-block-controls';
+
+/**
+ * Returns true if the block toolbar should be shown.
+ *
+ * @return {boolean} Whether the block toolbar component will be rendered.
+ */
+export function useHasBlockToolbar() {
+	const hasAnyBlockControls = useHasAnyBlockControls();
+	return useSelect(
+		( select ) => {
+			const {
+				getBlockEditingMode,
+				getBlockName,
+				getSelectedBlockClientIds,
+			} = select( blockEditorStore );
+
+			const selectedBlockClientIds = getSelectedBlockClientIds();
+			const selectedBlockClientId = selectedBlockClientIds[ 0 ];
```

Comment:
Suggestion: We can probably use `getBlockSelectionStart` here if we want only the first block clientId.

Reply:
I do think that would be a good change because `getBlockSelectionStart()` returns a state value and `selectedBlockClientIds()` does quite a bit more work. However, if I:

- Add three paragraphs
- click and drag from the third paragraph up to the first
- `getBlockSelectionStart()` will return the third paragraph ID (where selection started) and  `getSelectedBlockClientIds()[0]` will return the first paragraph ID.

So, as it would be a functional change, I think we should do it in a follow-up.

Reply:
Follow-up at https://github.com/WordPress/gutenberg/pull/59450